File: src/01-easy/horse-racing-duals/solution.pl
chomp(my $n = <STDIN>);
my @h;
for (1..$n) { chomp(my $x = <STDIN>); push @h, $x + 0; }
@h = sort { $a <=> $b } @h;
my $min = $h[1] - $h[0];
for my $i (1..$#h-1) {
    my $d = $h[$i+1] - $h[$i];
    $min = $d if $d < $min;
}
print "$min\n";

File: src/01-easy/horse-racing-duals/solution.dart
import 'dart:io';
void main() {
    int n = int.parse(stdin.readLineSync()!);
    List<int> h = List.generate(n, (_) => int.parse(stdin.readLineSync()!));
    h.sort();
    int min = h[1] - h[0];
    for (int i = 1; i < n - 1; i++) {
        int d = h[i+1] - h[i];
        if (d < min) min = d;
    }
    print(min);
}
